packages/authenticator/amplify_authenticator/lib/src/state/authenticator_state.dart

```diff
@@ -139,7 +139,46 @@ class AuthenticatorState extends ChangeNotifier {
     notifyListeners();
   }
 
-  String _confirmationCode = '';
+  late String _confirmationCode;
+
+  Set<MfaType>? get allowedMfaTypes {
+    final state = _authBloc.currentState;
+
+    if (state is ContinueSignInWithMfaSelection) {
+      return state.allowedMfaTypes;
+    }
+    return null;
+  }
+
+  MfaType? get selectedMfaMethod => _selectedMfaMethod;
+
+  /// The value for the MFA selection form field
+  ///
+  /// This value will be used during confirm sign up with MFA selection
+  set selectedMfaMethod(MfaType? value) {
+    _selectedMfaMethod = value;
+    notifyListeners();
+  }
+
+  late MfaType? _selectedMfaMethod;
```

Comment:
Nullable fields should never be `late` - they are already `null` by default.

```suggestion
  MfaType? _selectedMfaMethod;
```

Reply:
This needs to be reset in `_resetAttributes`